### PR TITLE
docs: document creating pages in Oryx

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
- Moves routing, compositions and this new doc into a folder about pages. 
- We'll add layout to this folder going forward
- We've also moved a few pages in the side bare up

closes [HRZ-2157](https://spryker.atlassian.net/browse/HRZ-2157)

[HRZ-2157]: https://spryker.atlassian.net/browse/HRZ-2157?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  .create();
+```
+
+Selectors use the following syntax:
+
+- select a page with the `#` prefix, e.g. `#home-page`
+- select a component globally by `id`, e.g. `my-composition`
+- select components by `id` or `tag`, e.g. `oryx-product-title`
+- chain selects, using the dot notation, e.g. `#home-page.my-composition.oryx-product-title`
+- skip parts of the data, e.g. `#home-page.oryx-product-title`
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:151]: Not sure I get this point. What exactly we are skipping here?

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+Using this syntax gives you flexibility to apply changes in any page, or to very specific pages.
+
+### Merge strategies
+
+When you do not provide a merge `type`, the selected component is replaced by default. Alternative types can be configured in the `merge.type` field.
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:157]: Maybe let's add an example?

**Review by tolerants — CHANGES_REQUESTED** [review]

**Review by tolerants — DISMISSED** [review]

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+In Oryx, pages are essential building blocks of web applications. They represent different sections or views within your application and can be created using a data-driven approach. This approach allows you to define the composition and layout of pages using external data sources, making it easier to maintain, customize, and optimize your application.
+
+Oryx provides standard pages in application [presets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-presets.html), such as home, login page, search page, etc. Using presets gets you up and running fast. This documentation shows you how to provide custom pages or apply small customization on top of standard presetted pages.
+
+## Understanding Pages and Compositions
+
+Pages in Oryx are represented as [compositions](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-compositions.html), which are collections of components organized in a specific order. Compositions enable you to define the structure and layout of pages without hardcoding them in your application code. This separation of concerns makes your components more reusable and less tied to specific pages.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:14]: Shall we replace `concern` with `responsibility`?

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+Oryx provides standard pages, like home, login, or search page, in [application presets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-presets.html). Using presets gets you up and running fast. This document shows you how to provide custom pages or apply small customization on top of standard preset pages.
+
+## Understanding pages and compositions
+
+Pages in Oryx are represented as [compositions](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-compositions.html), which are collections of components organized in a specific order. Compositions enable you to define the structure and layout of pages without hardcoding them in your code. This separation of concerns makes your components more reusable and less tied to specific pages.
+
+Oryx leverages a data-driven approach for creating pages, letting you configure the composition and content of pages using external data sources. For the advantages and technical details, see [Compositions](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-compositions.html).
+
+## Non-data-driven approach
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:18]: I feel like there must be a word for it. Hardcoded? model-driven? qualitative? subjective?

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+- select a component globally by `id`, e.g. `my-composition`
+- select components by `id` or `tag`, e.g. `oryx-product-title`
+- chain selects, using the dot notation, e.g. `#home-page.my-composition.oryx-product-title`
+- skip parts of the component tree, e.g. `#home-page.oryx-product-title` rather than `#home-page.my-composition.oryx-product-title`
+
+Using this syntax gives you flexibility to apply changes in any page, or to very specific pages.
+
+### Merge strategies
+
+When you do not provide a merge `type`, the selected component is replaced by default. Alternative types can be configured in the `merge.type` field.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:157]: And what is the default way to replace a selected component?

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+| `replace` (default) | Replaces the selected element with the given content                                                                                                           |
+| `patch`             | Patches the selected component with the given component. This includes both the component options and content. The data is deep merged, expect for arrays.     |
+| `before`            | Adds the content before the selected component.                                                                                                                |
+| `after`             | Adds the content after the selected component                                                                                                                  |
+| `append`            | Adds the content after the last component of the composition components. If the selected component is not a composition, the custom component is not merged.   |
+| `prepend`           | Adds the content before the first component of the composition components. If the selected component is not a composition, the custom component is not merged. |
```
Comment (supproduction) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:193]: Should we also mention about `remove` and possibility of using `refs`?

File: docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+  )
+  .create();
+```
+
+The table below gives an overview of the various merge types.
+
+| Strategy            | details                                                                                                                                                        |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `replace` (default) | Replaces the selected element with the given content                                                                                                           |
+| `patch`             | Patches the selected component with the given component. This includes both the component options and content. The data is deep merged, expect for arrays.     |
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md:189]:
```suggestion
| `patch`             | Patches the selected component with the given component. This includes both the component options and content. The data is deep merged, except for arrays.     |
| `remove`             | Removed selected component.    |
```

[the remark this answers is not in the document]
Comment (tolerants) [reply docs/scos/dev/front-end-development/202212.0/oryx/building-pages/oryx-pages.md]: There was a typo. `expect` => `except`

**Review by tolerants — DISMISSED** [review]

**Review by andriitserkovnyi — APPROVED** [review]

**Pull request merged**